src/sentry/api/bases/external_actor.py

```diff
@@ -127,6 +152,13 @@ def validate_team_id(self, team_id: int) -> Team:
         except Team.DoesNotExist:
             raise serializers.ValidationError("This team does not exist.")
 
+    def to_internal_value(self, data: Mapping[str, Any]) -> OrderedDict[str, Any]:
+        internal_value: OrderedDict[str, Any] = super().to_internal_value(data)
+        team_id = internal_value["team_id"]
+        if isinstance(team_id, Team):
+            internal_value["team"] = team_id
```

Comment:
Why is `team_id` an instance of `Team`? Shouldn't it be an int?

Reply:
This seems to be the default behavior with `rest_framework.serializers.ModelSerializer` which says that 
> The process of automatically determining a set of serializer fields based on the model fields is reasonably complex ...

See: https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L864-L879

Perhaps the default behavior can be changed with `ExternalTeamSerializer.Meta.fields`, but those fields are still necessary before backfilling all records with the team/user.